### PR TITLE
Add comprehensive test suite for Q CLI provider

### PR DESCRIPTION
**Pull request by gutosantos82**
## Summary
Adds complete test coverage for the Q CLI provider with 51 tests achieving 100% code coverage.

## What's Included
- **42 unit tests**: Fast, mocked tests covering initialization, status detection, message extraction, handoff scenarios, and edge cases
- **9 integration tests**: End-to-end tests with real Q CLI operations
- **Test fixtures**: Realistic terminal output with ANSI codes for all Q CLI states
- **Documentation**: Comprehensive README with usage examples and CI/CD integration

## Running Tests
```bash
# Unit tests (fast, <1s)
pytest test/providers/test_q_cli_unit.py -v

# Integration tests (requires Q CLI)
pytest test/providers/test_q_cli_integration.py -v

### CODE REVIEW

File: test/providers/README.md [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+**Agent Setup:**
+The integration tests automatically create a test agent named `developer` if it doesn't exist. The agent is created at:
+- `~/.aws/amazonq/cli-agents/developer.json`
+
+If you want to create the agent manually before running tests:
+```bash
+mkdir -p ~/.aws/amazonq/cli-agents
+cat > ~/.aws/amazonq/cli-agents/developer.json << 'EOF'
+{
+  "name": "developer",
```
Comment (tuanknguyen) [inline test/providers/README.md]: nit: could we pick an agent name related to "test" instead of "developer"?

File: test/providers/README.md [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+EOF
+```
+
+For more information on custom agents, see: https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-custom-agents.html
+
+## Running Tests
+
+### Run All Unit Tests (Recommended)
+```bash
+pytest test/providers/test_q_cli_unit.py -v
```
Comment (tuanknguyen) [inline test/providers/README.md]: It's probably better to have a DEVELOPMENT.md to describe the steps for development. One main change is to follow the uv convention for package management and virtual env management. Here's the brief version

In this PR, you can add `pytest` to the pyproject.toml using. This will add pytest to the `dev` group.
```
uv add --dev pytest
```

In DEVELOPMENT.md, we can add testing instructions:
```
git clone https://github.com/awslabs/cli-agent-orchestrator.git
cd cli-agent-orchestrator/
uv sync
```
(`uv sync` will install all necessary dependencies, including the `dev` group)

To run all test:
```
uv run pytest -v
```

To run a single test:
```
uv run pytest <file_name.py>
```
(and all other variations you listed below. The main difference is to prepend all commands with `uv run  ...`)

File: test/providers/README.md [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+- **q_cli_permission_output.txt** - Permission request prompt
+- **q_cli_error_output.txt** - Error message output
+- **q_cli_complex_response.txt** - Multi-line response with code blocks
+- **q_cli_handoff_successful.txt** - Successful handoff between agents
+- **q_cli_handoff_error.txt** - Failed handoff with error message
+- **q_cli_handoff_with_permission.txt** - Handoff requiring user permission
+
+## CI/CD Integration
+
+### GitHub Actions Example
```
Comment (tuanknguyen) [inline test/providers/README.md]: great idea! I don't see the change to `.github/workflows/` in this PR. Please add it

File: test/providers/README.md [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+        # Cleanup
+        tmux_client.kill_session(test_session_name)
+```
+
+## Troubleshooting
+
+### Unit Tests Fail with Import Error
+```bash
+# Install package in development mode
+pip install -e .
```
Comment (tuanknguyen) [inline test/providers/README.md]: `uv sync` and `uv run pytest ...` solve this issue. Another reason to use `uv` convention throughout.
